lib/model/rwfolder.go

```diff
@@ -1372,12 +1372,33 @@ func (f *rwFolder) dbUpdaterRoutine() {
 	tick := time.NewTicker(maxBatchTime)
 	defer tick.Stop()
 
+	fsync := os.Getenv("STFSYNC") != ""
+	var changedFiles []string
+	var changedDirs []string
+	if fsync {
+		changedFiles = make([]string, 0, maxBatchSize)
+		changedDirs = make([]string, 0, maxBatchSize)
+	}
+
 	handleBatch := func() {
 		found := false
 		var lastFile protocol.FileInfo
 
 		for _, job := range batch {
 			files = append(files, job.file)
+			if fsync {
+				// collect changed files
+				if job.jobType == dbUpdateHandleFile || job.jobType == dbUpdateShortcutFile {
+					changedFiles = append(changedFiles, filepath.Join(f.dir, job.file.Name))
+				}
+				// collect changed dirs
+				if job.jobType == dbUpdateHandleDir {
+					changedDirs = append(changedDirs, filepath.Join(f.dir, job.file.Name))
+				} else if job.jobType != dbUpdateShortcutFile {
+					changedDirs = append(changedDirs,
+						filepath.Dir(filepath.Join(f.dir, job.file.Name)))
```

Comment:
It can't be more than ``maxBatchSize``. I thought that it's faster to just add all and sort them once at the end.